src/parallel/GeneralDomainDecomposition.cpp

```diff
@@ -13,26 +13,38 @@
 #include "NeighbourCommunicationScheme.h"
 
 GeneralDomainDecomposition::GeneralDomainDecomposition(double interactionLength, Domain* domain)
-	: _boxMin{0.}, _boxMax{0.} {
-	std::array<double, 3> domainLength = {domain->getGlobalLength(0), domain->getGlobalLength(1),
-										  domain->getGlobalLength(2)};
-
-	auto gridSize = getOptimalGrid(domainLength, this->getNumProcs());
+	: _boxMin{0.},
+	  _boxMax{0.},
+	  _domainLength{domain->getGlobalLength(0), domain->getGlobalLength(1), domain->getGlobalLength(2)},
+	  _interactionLength{interactionLength} {}
+
+void GeneralDomainDecomposition::initializeALL() {
+	global_log->info() << "initializing ALL load balancer..." << std::endl;
+	auto gridSize = getOptimalGrid(_domainLength, this->getNumProcs());
 	auto gridCoords = getCoordsFromRank(gridSize, _rank);
-	_coversWholeDomain = {gridSize[0] == 1, gridSize[1] == 1, gridSize[2] == 1};
 	global_log->info() << "gridSize:" << gridSize[0] << ", " << gridSize[1] << ", " << gridSize[2] << std::endl;
 	global_log->info() << "gridCoords:" << gridCoords[0] << ", " << gridCoords[1] << ", " << gridCoords[2] << std::endl;
-	std::tie(_boxMin, _boxMax) = initializeRegularGrid(domainLength, gridSize, gridCoords);
+	std::tie(_boxMin, _boxMax) = initializeRegularGrid(_domainLength, gridSize, gridCoords);
+	if (_gridSize) {
```

Comment:
```suggestion
	if (_gridSize.has_value()) {
```
Would be a bit more readable, especially if one does not expect this member to be a `std::optional` but doesn't really matter... If you change it though, change it everywhere.